### PR TITLE
Bignum merge scaffolding

### PR DESCRIPTION
**Pull request by yanesca**
## Description

A lot of parallel work is going on on the new bignum files. Most of this is adding new functions and tests and result in trivial conflicts.

The goal of this PR is to avoid these trivial conflicts, allow more parallel work, save time and CI churn.

The idea is that each issue would take a slot and each PR belonging to the same issue would use the same slot. This should work as PRs implementing the same issue are normally consecutive. The last PR in a series implementing an issue would remove the slot separators.

## Gatekeeper checklist

- [x] **changelog** provided, or not required
- [x] **backport** done, or not required
- [x] **tests** provided, or not required

### CODE REVIEW

File: tests/suites/test_suite_bignum_core.misc.data [hunk 1]
```diff
@@ -364,3 +364,83 @@ mpi_core_get_mont_r2_unsafe:"8335616aed761f1f7f44e6bd49e807b82e3bf2bf11bfa63":"5
...
+
+BEGIN MERGE SLOT 1
+depends_on:MBEDTLS_BIGNUM_MERGE
+mpi_core_io_null
+
+END MERGE SLOT 1
+depends_on:MBEDTLS_BIGNUM_MERGE
+mpi_core_io_null
```
Comment (tom-cosgrove-arm) [inline tests/suites/test_suite_bignum_core.misc.data:374]: won't these two lines confuse the merge? I thought the idea was that the BEGIN/END lines needed to be the first/last of the "slot"...

Comment (yanesca) [reply tests/suites/test_suite_bignum_core.misc.data:374]: I didn't know about the comment syntax in .data files, with comments one line will be enough.

File: tests/suites/test_suite_bignum_mod_raw.function [hunk 2]
```diff
@@ -279,4 +279,44 @@ exit:
...
+
+/* END MERGE SLOT 8 */
+
+/* BEGIN MERGE SLOT 9 */
+
+/* END MERGE SLOT 9 */
+
+/* BEGIN MERGE SLOT 10 */
+
+/* END MERGE SLOT 10 */
```
Comment (tom-cosgrove-arm) [inline tests/suites/test_suite_bignum_mod_raw.function:321]: sorry, missed this earlier, but in the other `.function` files the merge slots are after the final `/* END_CASE */`, not before it - not sure if this matters...

Comment (yanesca) [reply tests/suites/test_suite_bignum_mod_raw.function:321]: Yep, it does matter, thanks for catching it.

**Review by tom-cosgrove-arm — DISMISSED** [review]
LGTM

Note there are blank lines at the end of:

```
tests/suites/test_suite_bignum_core.function
tests/suites/test_suite_bignum_core.misc.data
tests/suites/test_suite_bignum_mod.data
tests/suites/test_suite_bignum_mod_raw.data
```

**Review by wernerlewis — APPROVED** [review]
LGTM

**Comment by yanesca** [comment]
Rebased to resolve conflicts.

**Review by tom-cosgrove-arm — DISMISSED** [review]
LGTM

File: scripts/mbedtls_dev/bignum_mod_raw.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABCMeta
+
+from . import test_data_generation
+
+class BignumCoreTarget(test_data_generation.BaseTarget, metaclass=ABCMeta):
```
Comment (minosgalanakis) [inline scripts/mbedtls_dev/bignum_mod_raw.py:21]:
```suggestion
class BignumModRawTarget(test_data_generation.BaseTarget, metaclass=ABCMeta):
```

File: scripts/mbedtls_dev/bignum_mod.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABCMeta
+
+from . import test_data_generation
+
+class BignumCoreTarget(test_data_generation.BaseTarget, metaclass=ABCMeta):
```
Comment (minosgalanakis) [inline scripts/mbedtls_dev/bignum_mod.py:21]:
```suggestion
class BignumModTarget(test_data_generation.BaseTarget, metaclass=ABCMeta):
```

File: scripts/mbedtls_dev/bignum_mod_raw.py (hunk 3 above)
Comment (yanesca) [reply scripts/mbedtls_dev/bignum_mod_raw.py]: Nice catch, thanks!

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by wernerlewis — APPROVED** [review]

**Comment by yanesca** [comment]
Internal CI passed and the failure on OpenCI is unrelated.

**Comment by gilles-peskine-arm** [comment]
Registry of merge slots (please keep this updated!)

| Number | Owner | Content |
| ------ | ----- | ------- |
| 1 | @yanesca | mod_exp |
| 2 |@gabor-mezei-arm | mod sub |
| 3 | @tom-cosgrove-arm | sub int |
| 4 |  |  |
| 5 | @wernerlewis | add |
| 6 |  |  |
| 7 | @minosgalanakis | Montgomery conversions |
| 8 |  |  |
| 9 |  |  |

**Pull request merged**